### DISCUSSION TITLE
How to Let Uptime Kuma check status page?

### DISCUSSION THREAD

**Discussion by masterwishx** [authelia/authelia#3497] [Q&A] [answered]
How to configure Authelia for let UptimeKuma
to check page status by GET?
Using Nginx Proxy Manager.
But when Kuma checking its redirect to auth.mysite.com instead if photo.mysite.com
Becouse of setting in NPM. 
Can I config authelia somehow or I need config nginx for it?

**Comment by james-d-elliott** [comment]
You can create a rule to either bypass authentication for traffic coming fro UptimeKuma based on IP address, or configure a one_factor policy for a service user and pass the credentials from UptimeKuma  to Authelia via the `Proxy-Authorization` header using basic auth.

**Reply by masterwishx** [reply]
> You can create a rule

The problem that  I'm using cloudflare proxy
So Ip not same every time. 
But I can use they proxy ip's list for one password rule don't know if it's good idea.
Also don't have much knowledge in nginx
Maybe I can bypass somehow by client name like Kuma in authelia or nginx

**Reply by james-d-elliott** [reply]
What do you mean client name?

**Reply by masterwishx** [reply]
"Uptime-Kuma/1.18.5" "-" The end in NPM log

![image](https://user-images.githubusercontent.com/28630321/209448507-71f58bdd-e4b0-4deb-90de-dd6e7ba27373.png)

**Comment by Normanras** [comment]
To jump in on this, I was just trying something like this today and I'm also using NPM. When I add the custom code to the Advanced Tab to direct it to the Authelia custom `conf` files, I get a Websockets error with Uptime Kuma. 

As soon as I take that advanced config away, the error goes away.

I don't know much about Websockets and how Authelia interacts with them, but I guess I should start doing some research...

**Reply by james-d-elliott** [reply]
There should be some logs in both nginx and authelia which correlate to that issue

**Reply by Normanras** [reply]
I'll check it out. Thanks

**Reply by Normanras** [reply]
Here's the entry. 
```
time="2022-12-21T14:57:45Z" level=debug msg="Check authorization of subject username=norm groups=my_group ip=123.123.12.12 and object https://status.mydomain.com/socket.io/?EIO=4&transport=websocket (method GET)."
```

Here's how I fixed it! Advanced Tab in your Nginx-Proxy-Manager Proxy Host:
```
include /config/nginx/snippets/authelia-location.conf;

location / {
    include /config/nginx/snippets/proxy.conf;
    include /config/nginx/snippets/authelia-authrequest.conf;
    proxy_pass $forward_scheme://$server:$port;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "upgrade";
}
```

And more specifically, you need to add these two lines:
```
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "upgrade";
```

I hope this helps you @masterwishx ! @james-d-elliott, let me know if you want me to add this anywhere to Authelia's Docs.

**Reply by masterwishx** [reply]
@Normanras 
Thanks, i need to check it but im not using snippets in Nginx Proxy Manager. becose used IbraCorp Docs for install Authelia+NPM.

but after @james-d-elliott told me to use like in Authelia Docs (snippets included in NPM) i need to change the ALL my setup Authelia +NPM for it.

**Comment by james-d-elliott** [comment, marked as answer]
We also added bearer auth via client credentials grant. More information here: https://www.authelia.com/integration/openid-connect/oauth-2.0-bearer-token-usage/
